CRM/Civicase/Hook/PostProcess/CaseCategoryPostProcessor.php

```diff
@@ -25,53 +23,28 @@ public function run($formName, CRM_Core_Form &$form) {
       return;
     }
 
-    $caseCategoryMenu = new CaseCategoryMenuService();
-    $caseCategoryCustomData = new CaseCategoryCustomDataType();
-    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
-
+    // Get object data from submitted from.
     $formValues = $form->_submitValues;
-    $formAction = $form->getVar('_action');
+    $caseCategoryValues = $form->getVar('_values');
+    $category = [
+      'id' => $form->getVar('_id'),
+      'name' => !empty($caseCategoryValues['name']) ? $caseCategoryValues['name'] : $formValues['label'],
+      'is_active' => $formValues['is_active'],
+      'icon' => $formValues['icon'],
+    ];
```

Comment:
We really don't need an array here tbh since we are not passing this directly to a function. 
In this case, will be better to have them as variables.

```php
$categoryId = $form->getVar('_id');
$categoryName = !empty($caseCategoryValues['name']) ? $caseCategoryValues['name'] : $formValues['label'];
```

Reply:
I just like to have set of similar variables in array, and it does not violate any coding standards, right? So what is the difference?

Reply:
TBH no difference but why create an array when it is not needed?
An Array is meant to store a set of variables but in this case, do we need it? no